core/shared/src/main/scala/fs2/StreamLeg.scala

```diff
@@ -0,0 +1,36 @@
+package fs2
+
+import fs2.internal.{Algebra, CompileScope, FreeC}
+
+final class StreamLeg[F[_], O](
```

Comment:
Thoughts on defining this inside `Stream` companion? `Stream.Leg[F, O]`

Reply:
Also, can we add some ScalaDoc here?

Reply:
:+1: to `Stream.Leg`

Reply:
Good idea, will maek the chane + add the doc